Build failures due to static `allocate_key_context`

Disabling cryptography macros results in (for example)

```
/home/runner/work/libspdm/libspdm/os_stub/cryptlib_openssl/pem/pem_pqc.c:27:13: error: ‘allocate_key_context’ defined but not used [-Werror=unused-function]
   27 | static bool allocate_key_context(EVP_PKEY *pkey, void **context)
```
detected by GCC.

--- a/os_stub/cryptlib_openssl/der/der.c
+++ b/os_stub/cryptlib_openssl/der/der.c
@@ -1,6 +1,6 @@
 /**
  *  Copyright Notice:
- *  Copyright 2021-2022 DMTF. All rights reserved.
+ *  Copyright 2021-2026 DMTF. All rights reserved.
  *  License: BSD 3-Clause License. For full text see link: https://github.com/DMTF/libspdm/blob/main/LICENSE.md
  **/
 
@@ -13,6 +13,10 @@
 #include <openssl/x509.h>
 #include <openssl/evp.h>
 #include <openssl/decoder.h>
+
+#if (LIBSPDM_RSA_SSA_SUPPORT) || (LIBSPDM_RSA_PSS_SUPPORT) || (LIBSPDM_ECDSA_SUPPORT) || \
+    (LIBSPDM_EDDSA_ED25519_SUPPORT) || (LIBSPDM_EDDSA_ED448_SUPPORT) || \
+    (LIBSPDM_SM2_DSA_SUPPORT)
 
 /**
  * Helper function to retrieve public key from DER data using BIO.
@@ -69,6 +73,9 @@ static bool allocate_key_context(EVP_PKEY *pkey, void **context)
     *context = ctx;
     return true;
 }
+#endif /* (LIBSPDM_RSA_SSA_SUPPORT) || (LIBSPDM_RSA_PSS_SUPPORT) || (LIBSPDM_ECDSA_SUPPORT) || \
+          (LIBSPDM_EDDSA_ED25519_SUPPORT) || (LIBSPDM_EDDSA_ED448_SUPPORT) || \
+          (LIBSPDM_SM2_DSA_SUPPORT) */
 
 #if (LIBSPDM_RSA_SSA_SUPPORT) || (LIBSPDM_RSA_PSS_SUPPORT)
 /**

--- a/os_stub/cryptlib_openssl/pem/pem_pqc.c
+++ b/os_stub/cryptlib_openssl/pem/pem_pqc.c
@@ -1,6 +1,6 @@
 /**
  *  Copyright Notice:
- *  Copyright 2025 DMTF. All rights reserved.
+ *  Copyright 2025-2026 DMTF. All rights reserved.
  *  License: BSD 3-Clause License. For full text see link: https://github.com/DMTF/libspdm/blob/main/LICENSE.md
  **/
 
@@ -14,6 +14,8 @@
 #include <openssl/evp.h>
 
 int PasswordCallback(char *buf, const int size, const int flag, const void *key);
+
+#if (LIBSPDM_ML_DSA_SUPPORT) || (LIBSPDM_SLH_DSA_SUPPORT)
 
 /**
  * Helper function to allocate and initialize a key context wrapper.
@@ -36,6 +38,7 @@ static bool allocate_key_context(EVP_PKEY *pkey, void **context)
     *context = ctx;
     return true;
 }
+#endif /* (LIBSPDM_ML_DSA_SUPPORT) || (LIBSPDM_SLH_DSA_SUPPORT) */
 
 #if LIBSPDM_ML_DSA_SUPPORT
 

--- a/unit_test/test_crypt/x509_verify.c
+++ b/unit_test/test_crypt/x509_verify.c
@@ -1,6 +1,6 @@
 /**
  *  Copyright Notice:
- *  Copyright 2021-2025 DMTF. All rights reserved.
+ *  Copyright 2021-2026 DMTF. All rights reserved.
  *  License: BSD 3-Clause License. For full text see link: https://github.com/DMTF/libspdm/blob/main/LICENSE.md
  **/
 #include "test_crypt.h"
@@ -516,25 +516,33 @@ bool libspdm_validate_crypt_x509(char *Path, size_t len)
     asym_nid = libspdm_get_aysm_nid_from_file_name(Path, len);
 
     switch (asym_nid) {
+    #if LIBSPDM_RSA_SSA_SUPPORT
     case LIBSPDM_CRYPTO_NID_RSASSA2048:
     case LIBSPDM_CRYPTO_NID_RSASSA3072:
         status = libspdm_rsa_get_private_key_from_pem(
             test_private_key, test_private_key_len, NULL, &context);
         break;
+    #endif /* LIBSPDM_RSA_SSA_SUPPORT */
+    #if LIBSPDM_ECDSA_SUPPORT
     case LIBSPDM_CRYPTO_NID_ECDSA_NIST_P256:
     case LIBSPDM_CRYPTO_NID_ECDSA_NIST_P384:
         status = libspdm_ec_get_private_key_from_pem(
             test_private_key, test_private_key_len, NULL, &context);
         break;
+    #endif /* LIBSPDM_ECDSA_SUPPORT */
+    #if LIBSPDM_SM2_DSA_SUPPORT
     case LIBSPDM_CRYPTO_NID_SM2_DSA_P256:
         status = libspdm_sm2_get_private_key_from_pem(
             test_private_key, test_private_key_len, NULL, &context);
         break;
+    #endif /* LIBSPDM_SM2_DSA_SUPPORT */
+    #if LIBSPDM_EDDSA_SUPPORT
     case LIBSPDM_CRYPTO_NID_EDDSA_ED25519:
     case LIBSPDM_CRYPTO_NID_EDDSA_ED448:
         status = libspdm_ecd_get_private_key_from_pem(
             test_private_key, test_private_key_len, NULL, &context);
         break;
+    #endif /* LIBSPDM_EDDSA_SUPPORT */
     default:
         libspdm_my_print("\n  - Get Private Key - [Fail]");
         status = false;

